prql-compiler/tests/integration/connection.rs

```diff
@@ -29,17 +32,21 @@ pub struct MysqlConnection(pub mysql::Pool);
 pub struct MssqlConnection(pub tiberius::Client<Compat<TcpStream>>);
 
 pub trait DBConnection {
-    fn run_query(&mut self, sql: &str, runtime: &Runtime) -> Vec<Row>;
+    fn run_query(&mut self, sql: &str, runtime: &Runtime) -> Result<Vec<Row>>;
 
     fn import_csv(&mut self, csv_name: &str, runtime: &Runtime);
 
     fn get_dialect(&self) -> Dialect;
+
+    fn modify_sql(&self, sql: String) -> String {
```

Comment:
```suggestion
    // We sometimes want to modify the SQL `INSERT` query (we don't modify the
    // SQL `SELECT` query
    fn modify_sql(&self, sql: String) -> String {
```

Just to make clear that we're testing the `SELECT` without modification. We could also rename the method if we have the patience, but no great stress.

Reply:
That's not true. the `SELECT` is modified as well ([see here](https://github.com/PRQL/prql/blob/0153056deadae7a573c5f15f365c39bd487fdeda/prql-compiler/tests/integration/main.rs#L183)). It is necessary for casting. For example in MySQL the type `INT` exists, but to cast it to an int you need to use `SIGNED` :facepalm:

Reply:
Ah, good point, sorry for my misunderstanding. I see we changed from https://github.com/PRQL/prql/blob/712ae15296a6ba41092b7e870595489425bb1d54/prql-compiler/tests/integration/main.rs#L160-L164

I'm a bit concerned about this approach, because it hides the frailties of PRQL. It's possible to make the tests pass by adding things to this function, but PRQL is still arguably producing the incorrect SQL.

I had a look through the examples — they're mostly types — which aren't easy to get precise given the heterogeneity there. Maybe that's OK for the moment — these tests are valuable enough that we don't want to have to exclude everything. 

There's one that's quotes — that's more concerning — I think we should instead fix that if it's from PRQL query. https://github.com/PRQL/prql/blob/0153056deadae7a573c5f15f365c39bd487fdeda/prql-compiler/tests/integration/connection.rs#L278

I think two options from here:
1. Remove these replacements, and put the `CAST` functions in a single example and exclude that for DBs (but maybe for all DBs, in which case less attractive...)
2. Split up the replacements for `schema.sql` — which is totally fine to mangle — and for the queries themselves — and have `TODO`s around the later. Then in the long term we can think about how to eliminate them.

WDYT?

Reply:
I removed the replacement of the double quotes because it's not necessary. It was added, when there was table with an uppercase letter in the schema.sql. Btw, we should an integration test for that. It's a tricky subject.

I agree with you that with the current modifying of SQL some issues of the compiler can slip through.
I would prefer the first option.

Reply:
OK great — do you think we can do the first option? If we exclude `invoice_totals` from most of the dialects, can we remove these replacements from the query path?

Reply:
I removed the CASTs from invoice_totals.prql and added it to a separate file

Reply:
#2462 won't be necessary anymore